### PR TITLE
Add develop branch

### PR DESCRIPTION
**Pull request by quietsy**
Added a develop branch to nextcloud for "beta" releases.

The latest version is found using the following command:

```bash
curl -sX GET https://api.github.com/repos/nextcloud/server/releases \
      | jq -r '.[] | select(.prerelease == true) | .tag_name' \
      | sed 's|^v||g' | sort -rV | head -1
```

Then downloaded from the following location:

```https://download.nextcloud.com/server/prereleases/nextcloud-${NEXTCLOUD_RELEASE}.tar.bz2```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -82,12 +82,12 @@ RUN \
...
     NEXTCLOUD_RELEASE=$(curl -sX GET https://api.github.com/repos/nextcloud/server/releases \
-      | jq -r '.[] | select(.prerelease != true) | .tag_name' \
+      | jq -r '.[] | .tag_name' \
       | sed 's|^v||g' | sort -rV | head -1); \
   fi && \
   curl -o \
     /tmp/nextcloud.tar.bz2 -L \
-    https://download.nextcloud.com/server/releases/nextcloud-${NEXTCLOUD_RELEASE}.tar.bz2 && \
+    https://github.com/nextcloud/server/archive/refs/tags/v${NEXTCLOUD_RELEASE}.tar.gz && \
   tar xf /tmp/nextcloud.tar.bz2 -C \
```
Comment (thespad) [inline Dockerfile:91]: Should probably change the extensions to be consistent with the download to avoid confusing anyone.

File: readme-vars.yml [hunk 2]
```diff
@@ -20,6 +20,7 @@ available_architectures:
 development_versions: false
```
Comment (thespad) [inline readme-vars.yml:20]: This needs to be `true` or the table won't show up

File: Dockerfile (hunk 1 above)
Comment (quietsy) [reply Dockerfile]: Done, good catch

File: readme-vars.yml (hunk 2 above)
Comment (quietsy) [reply readme-vars.yml]: Done

File: jenkins-vars.yml [hunk 3]
```diff
@@ -3,12 +3,14 @@
...
-release_tag: latest
-ls_branch: master
+custom_version_command: "curl -sX GET https://api.github.com/repos/nextcloud/server/releases | jq -r '.[] | .tag_name' | sed 's|^v||g' | sort -rV | head -1"
+release_type: prerelease
+release_tag: develop
+ls_branch: develop
 build_armhf: false
 repo_vars:
+  - EXT_USER = 'nextcloud'
+  - EXT_REPO = 'nextcloud'
```
[the remark this answers is not in the document]
Comment (quietsy) [reply jenkins-vars.yml]: Done

**Review by nemchik — APPROVED** [review]
lgtm

**Pull request merged**